backend/src/server/plugins/primary-forwarding-mode.ts

```diff
@@ -0,0 +1,14 @@
+import replyFrom from "@fastify/reply-from";
+import fp from "fastify-plugin";
+
+export const primaryForwardingMode = fp(async (server, opt: { primaryUrl: string }) => {
```

Comment:
might be better to include something about forwarding writes so that it is more specific. like forward mutations/writes